### PR TITLE
Add documentation for shadowing support.

### PR DESCRIPTION
**Pull request by nickygerritsen**
These docs reflect the changes in #756.

### CODE REVIEW

File: doc/manual/shadow.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+best to not have any other data for the contest like teams, team categories,
+team affiliations, submissions and clarifications. The easiest way to create the
+contest is to import it. For this you need a ``contest.yaml`` and a
+``problemset.yaml`` as described in the `ICPC CCS requirements page
+<https://clics.ecs.baylor.edu/index.php?title=Contest_Control_System_Requirements>`_.
+Furthermore you need problem directories as described on the
+:doc:`problem format <problem-format>` page. Place the two YAML files and all
+directories together in one directory and use the
+``misc-tools/import-contest.sh`` script to import them. Do not try to import
+groups, t eams or accounts using it, but do import all other things. Note that
```
Comment (thijskh) [inline doc/manual/shadow.rst:50]: extra space

File: doc/manual/shadow.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
+Appendix: Running DOMjudge as a shadow system
+=============================================
+
+It is possible to run DOMjudge as a shadow system behind another system.
+
+Shadowing means that DOMjudge will read events from an external system and mimic
+all actions described in those events. This is useful if one wants to verify if
+the results of an external system match DOMjudge, for example to verify both
+systems give the same judging results.
```
Comment (eldering) [inline doc/manual/shadow.rst:9]: Nice to add that this is used at the ICPC World Finals.

File: doc/manual/shadow.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+After doing all above steps, you can run the event feed import command to start
+importing events from the primary system. From the DOMserver installation
+directory, run::
+
+  import/import-event-feed key-from-yaml
+
+Where ``key-from-yaml`` is the key you configured in the previous step. The
+command will start running and import events.
+
+If the command looses its connection to the primary system it will automatically
```
Comment (eldering) [inline doc/manual/shadow.rst:82]: Typo: should be `loses`.

File: doc/manual/shadow.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+reconnect and continue reading where it left off. To stop the command, press
+``ctrl-c``. If you rerun the command, it will start again where it left off.
+
+If you ever want to restart reading the feed from the beginning (for example
+when the primary system had an error and fixed that), you can pass
+``--from-start`` to the command to start reading from the beginning. Note that
+the way DOMjudge keeps track of the events that it already processed is by
+keeping a copy of the event feed in ``TMPDIR`` as defined in
+``/etc/domserver-static.php``. This feed is named after the key you configured
+in the YAML file, so if you ever reuse the key, make sure to remove the caced
```
Comment (eldering) [inline doc/manual/shadow.rst:92]: `cached`

File: doc/manual/shadow.rst [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+Viewing shadow differences
+--------------------------
+
+In the admin interface, the *Shadow Differences* page will show all differences
+between the primary system and DOMjudge. Note that it will only show differences
+in the final verdict, not on a testcase level.
+
+At the top is a matrix where a summary of the differences is shown. All red
+cells are actual differences. If they appear in the *JU* column or row, this
+simply means one of the systems is not done yet with processing this submission.
```
Comment (eldering) [inline doc/manual/shadow.rst:104]: Maybe change `processing` to `judging` to make the meaning of `JU` clearer?

**Comment by valipanahinejhad** [comment]
خوب وکاربردی

**Pull request merged**